Update Handsontable to 7.1.0

trshafer opened the pull request:
I validated that this works with the rhandsontable_corr example.

Fixes #317

jrowen commented:
@trshafer I really appreciate you taking the initiative to put the PR together and drive the licensing discussion forward.  I've somewhat moved on to other projects recently, and this library hasn't received the attention it deserves.  Let me know if you're interested in remaining involved, and we can discuss how to collaborate going forward.

trshafer commented:
Hey @jrowen, thanks for your kind words. I'm interested in pushing the license conversation forward and am happy to do this work to update this package to the most up to date handsontable version, potentially supporting multiple versions if necessary. I'm away next week, but will likely be able come back to this the following week.

However, I am not interested in actively maintaining this package. I don't use `R` in my day job and that will make it difficult for me to remain dedicated to the proper maintenance of this package going forward. I think if you have moved on from being able to dedicate time to this project, which I can totally understand, it makes sense to add a maintainers needed section to the Readme.

trshafer commented:
Hey @jrowen, I have added the LICENSE file along with the source as requested by https://github.com/handsontable/handsontable/issues/6110#issuecomment-512821744. I have also updated the Readme and License.note file to reflect the change in the license. The Readme now provides instructions for versioning the installation of rhandsontable.

himanga commented:
For what it's worth, I really like the idea of at least keeping a version of this project available for free commercial use, which I think means doing one of the following:

- Stick with version 6.x
- Maintain multiple versions of this package, not sure how to implement this except in separate repositories.
- Separate the installation of this package from the handsontable

The last option would be most difficult for users, while the second option sounds most difficult for maintainers.

trshafer commented:
@himanga Do you feel the workaround proposed in the readme would work? https://github.com/jrowen/rhandsontable/pull/318/files#diff-04c6e90faac2675aa89e2176d2eec7d8

himanga commented:
@trshafer, I suspect that most packages that are on CRAN are used without thinking about licensing. Not to say users should't look at the readme, but I think the default installation should not lead people to use code incorrectly. Maybe asking users to pass the phrase 'non-commercial-and-evaluation' themselves is enough to prevent that, but now the default installation wouldn't be functional.  What if both packages were installed, and the user had the option to switch to version 7.x if they pass a license key or the non-commercial phase?

jrowen commented:
I believe @timelyportfolio is also working on a solution here that would support multiple versions of the underlying JS library.

sclewis23 commented:
Is this still a possibility ?  seems like other wrappers are taking same approach by indicating a license key is required for commercial use.
